addons/mail/static/src/models/activity_view/activity_view.js

```diff
@@ -63,13 +68,88 @@ registerModel({
         onClickUploadDocument() {
             this.fileUploaderView.component.openBrowserFileUploader();
         },
+        /**
+         * @private
+         * @returns {String|FieldCommand}
+         */
+        _computeAssignedUserText() {
+            if (!this.activity.assignee) {
+                return clear();
+            }
+            return _.str.sprintf(this.env._t("for %s"), this.activity.assignee.nameOrDisplayName);
+        },
+        /**
+         * @private
+         * @returns {String|FieldCommand}
+         */
+        _computeDelayLabel() {
+            if (!this.activity.dateDeadline) {
+                return clear();
+            }
+            const today = moment().startOf('day');
+            const momentDeadlineDate = moment(auto_str_to_date(this.activity.dateDeadline));
+            // true means no rounding
+            const diff = momentDeadlineDate.diff(today, 'days', true);
+            if (diff === 0) {
+                return this.env._t("Today:");
+            } else if (diff === -1) {
+                return this.env._t("Yesterday:");
+            } else if (diff < 0) {
+                return _.str.sprintf(this.env._t("%d days overdue:"), Math.abs(diff));
+            } else if (diff === 1) {
+                return this.env._t("Tomorrow:");
+            } else {
+                return _.str.sprintf(this.env._t("Due in %d days:"), Math.abs(diff));
+            }
+        },
```

Comment:
This one is actually wrong as a cached value, because it will fail to update when time changes.

It was more acceptable when a render would always call the method (even though the value would also not update if there was no render, at least there was a way, which was likely to happen, to unblock it).

The proper solution would be to add a model for time, and it should update its fields at a frequency we consider acceptable. And then this compute should depend on the fields of the time record.